WordPress/wordpress-playground
# wp now: activate plugin and theme the first time

- Closes https://github.com/WordPress/wordpress-playground/issues/238

## What?



- Add plugin and theme auto activation the first time running a project in plugin or theme mode.

## Why?



It helps few steps for the users running this mode.

## How?



I needed to change native `DOMParser` to use a library `jsdom` , because NodeJS doesn't come with parser.

## Testing Instructions





- Copy the path to a plugin or theme directory.
- Make sure is the first time you have run this project or remove the folder `~/.wp-now/wp-content/${projectFolder}`
- Run `npx nx preview wp-now start --path=/path/to/your/plugin-or-theme`
- Observe the WordPress homepage opens.
- Go to wp-admin > themes or plugins
- Observe your plugin or theme is activated.
